hazelcast/test/CMakeLists.txt

```diff
@@ -1,38 +1,43 @@
 cmake_minimum_required(VERSION 3.10)
 
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     #required to suppress the warning:
     # `gtest-death-test.cc:1301:24: error: 'dummy' may be used uninitialized [-Werror=maybe-uninitialized]`
     #for googletest source compilation
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wno-maybe-uninitialized")
 endif ()
 
-find_package(GTest QUIET)
```

Comment:
What about using the same method as you used for thrift, try find and use the found one or otherwise download. You may be able to solve the proper lib naming it with a variable.

Reply:
Done.